janitor.go

```diff
@@ -0,0 +1,134 @@
+package fscache
+
+import (
+	"sort"
+	"time"
+)
+
+type janitorKV struct {
+	Key   string
+	Value FileStream
+}
+
+// Janitor is used to control when there are too many streams
+// or the size of the streams is too big.
+// It is called once right after loading, and then it is run
+// again after every Next() period of time.
+type Janitor interface {
+	// Returns the amount of time to wait before the next scheduled Reaping.
+	Next() time.Duration
+
+	// Given a key and the last r/w times of a file, return true
+	// to remove the file from the cache, false to keep it.
```

Comment:
nit: Comment no longer matches the function.

But more generally, shouldn't we just rename the Janitor to LRUHaunter or something, and make it impl. Haunter directly now?

Reply:
\+

Reply:
^ Does this mean you think we should still call it Janitor?